[docs] redesign: new Error pages

Simek opened the pull request:
# Why

This PR adds new Error pages which are a part of docs redesign effort.

# How

Error pages will be the first pages which use new `<Header>` component, and they will replace the old ones immediately after merge.

The design and images have been taken from the Figma mocks, however, it looks like that currently the background colors of Header and Content are flipped in comparison to the mock (at least in dark mode), but this might be a side-effect of latest Expo Styleguide changes. I have also stick to the primary Button color, in comparison to the black/white button in mock, to leave a bit of color on the overall almost grayscale page.

CC @jonsamp It would be nice if you can confirm that the colors are correct. 🙂 

# Test Plan

The updated error pages have been tested by running docs website on `localhost`.

# Preview

## Desktop

#### Redirect
<img width="1261" alt="Screenshot 2022-04-10 at 13 11 49" src="https://user-images.githubusercontent.com/719641/162616685-b28ac9fb-d6f8-43b4-b4ef-d1e2ea73eab5.png">

#### 404
<img width="1261" alt="Screenshot 2022-04-10 at 13 08 34" src="https://user-images.githubusercontent.com/719641/162616688-15976c0e-ba8c-46bb-8eb7-9e518a463c6a.png">

#### 404 Redirected
<img width="1261" alt="Screenshot 2022-04-10 at 13 09 03" src="https://user-images.githubusercontent.com/719641/162616686-d7e39a80-11ae-4972-bc8a-f24f778e2b66.png">

## Mobile

#### Redirected
![image](https://user-images.githubusercontent.com/719641/162616971-b2155635-dd7d-493c-a33d-b107b997ddef.png)

#### 404
![image](https://user-images.githubusercontent.com/719641/162616984-518e4d92-b541-49bd-9f0c-fc34ea6cd12f.png)

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

Simek committed:
[docs] redesign: new Error pages

Simek committed:
small CSS tweak

Simek committed:
fix failing test-links, fix error pages title

Simek committed:
code cleanup

jonsamp commented:
![redirect](https://user-images.githubusercontent.com/6455018/162791298-3bcc2f05-ace6-4553-8a4e-bf44d89fc7c9.svg)
![server-error](https://user-images.githubusercontent.com/6455018/162791300-6cd2e692-697e-4dc8-9ae2-3ef4edac4012.svg)
![not-found](https://user-images.githubusercontent.com/6455018/162791301-6b352c57-f2e4-4867-82c5-aa1a9aa4d5d6.svg)
[svgs.zip](https://github.com/expo/expo/files/8466225/svgs.zip)

Can we replace those illustrations with these? They should be easier to style. Also, I'm hoping that the outline of the "page" part of the illustration should be our border color, and the page itself should have a default background color, while the centered icon should be our icon color.

Simek committed:
update design, alter images based on theme

Simek committed:
fix for the page background color alteration

Simek commented:
I have updated the design of the pages to match the latest mocks and I have added the provided images as components, so we can alter them using theme variables.

# Preview
<img width="1919" alt="Screenshot 2022-04-11 at 21 49 51" src="https://user-images.githubusercontent.com/719641/162818286-fc1c2b5f-4f73-4be2-a164-18e496629bed.png">

<img width="1919" alt="Screenshot 2022-04-11 at 21 52 06" src="https://user-images.githubusercontent.com/719641/162818648-c7a5cb5e-d3ff-4d20-ac3a-d13d0b369a3a.png">

Simek committed:
remove static images
